core/src/test/java/hudson/model/RunTest.java

```diff
@@ -262,4 +265,52 @@ public void compareRunsFromDifferentParentsWithSameNumber() throws Exception {
         assertTrue(r1.compareTo(r2) != 0);
         assertTrue(treeSet.size() == 2);
     }
+
+    @Test
+    public void willTriggerLogToStartWithNextFullLine() throws Exception {
+        assertWriteLogToEquals("Sample build output 3.\nSample build output 4.\nFinished: SUCCESS.\n", 2 * 23 + 10);
```

Comment:
Could you create constants for these magic numbers? It's not easy to understand they are related with the number of lines and characters.

Reply:
Sure thing. I put the whole sample output file to a constant and used its `length` where possible. I think it should be clear now.